client/components/ManageTestQueue/index.jsx

```diff
@@ -730,14 +1371,141 @@ const ManageTestQueue = ({
+                                                {ats.map(item => {
+                                                    return (
+                                                        <option
+                                                            key={item.id}
+                                                            value={item.id}
+                                                        >
+                                                            {item.name}
+                                                        </option>
+                                                    );
+                                                })}
+                                            </Form.Select>
+                                        )}
+                                    </Form.Group>
+                                    <Form.Group>
+                                        <Form.Label>Browser</Form.Label>
+                                        <Form.Select
+                                            value={updateBrowserSelection}
+                                            onChange={handleBrowserChange}
+                                            required
+                                        >
+                                            <option value={''} disabled>
+                                                Select a Browser
+                                            </option>
+                                            {ats
+                                                .find(
+                                                    at =>
+                                                        at.id ===
+                                                        updateAtSelection
+                                                )
+                                                ?.browsers.map(item => (
+                                                    <option
+                                                        key={`${item.name}-${item.id}`}
+                                                        value={item.id}
+                                                    >
+                                                        {item.name}
+                                                    </option>
+                                                ))}
+                                        </Form.Select>
+                                    </Form.Group>
+                                </Row>
+                            ) : null}
+                        </ModalInnerSectionContainer>
+                    }
+                    actionLabel={actionButtonLabel}
+                    handleAction={() => {
+                        if (actionButtonLabel === 'Save Changes') {
+                            runMutationForRequiredReportTable(
+                                'updateRequiredReport'
+                            );
+                            setUpdateAtSelection(
```

Comment:
This and the conditional below it appear in both of the higher level conditionals. Put them at the end of `handleAction` to avoid repetition.

Reply:
Ok. Got it. Changes made.